libvirt/resource_libvirt_domain.go

```diff
@@ -996,3 +1048,56 @@ func getDomainInterfaces(domain *libvirt.VirDomain) ([]libvirt.VirDomainInterfac
 
 	return interfaces, nil
 }
+
+func getDomainInterfacesFromNetworks(domain defDomain, meta interface{}) ([]libvirt.VirDomainInterface, error) {
```

Comment:
This function is really similar to [`getDomainInterfacesViaQemuAgent`](https://github.com/dmacvicar/terraform-provider-libvirt/blob/master/libvirt/qemu_agent.go#L32). It is a pity they do not have the same function signature...

Shouldn't we invoke it from [inside `getDomainInterfaces`](https://github.com/dmacvicar/terraform-provider-libvirt/blob/master/libvirt/resource_libvirt_domain.go#L979)? So we could try to get the interfaces with the qemu agent, then use `getDomainInterfacesFromNetworks` if it fails, and then try with `ListAllInterfaceAddresses` as a last resource...